core/src/main/java/org/kohsuke/stapler/bind/BoundObjectTable.java

```diff
@@ -50,6 +55,88 @@
  * @author Kohsuke Kawaguchi
  */
 public class BoundObjectTable implements StaplerFallback {
+
+    private static boolean isValidJavaScriptIdentifier(String variableName) {
+        // Ultimately this will be used as a JS identifier, so we need (a subset of) what's valid there.
+        // The primary purpose of this check however is to prevent injection attacks.
+        return variableName.matches("^[a-zA-Z_][a-zA-Z0-9_]*$");
+    }
+
+    private static boolean isValidJavaIdentifier(String name) {
+        if (name == null || StringUtils.isBlank(name)) {
+            return false;
+        }
+        if (!Character.isJavaIdentifierStart(name.charAt(0))) {
+            return false;
+        }
+        return name.substring(1).chars().allMatch(Character::isJavaIdentifierPart); // TODO More restrictive
+    }
+
+    /**
+     * This serves the script content for a bound object. Support CSP-compatible st:bind and similar methods of making
+     * objects accessible to JS.
+     *
+     * @param req
+     * @param rsp
+     * @param var the variable name to assign the Stapler proxy to
+     * @param methods the list of methods (needed for {@link WithWellKnownURL})
+     * @throws IOException
+     */
+    public void doScript(StaplerRequest req, StaplerResponse rsp, @QueryParameter String var, @QueryParameter String methods) throws IOException {
+        final String boundUrl = req.getRestOfPath();
+
+        if (var == null) {
+            return;
+        }
+
+        if (!isValidJavaScriptIdentifier(var)) {
+            LOGGER.log(Level.FINE, () -> "Rejecting invalid JavaScript identifier: " + var);
+            return;
+        }
```

Comment:
Breaks https://github.com/jenkinsci/build-monitor-plugin/blob/e07bbdfc6dac962ed030266aa556305f580100c0/build-monitor-plugin/src/main/resources/com/smartcodeltd/jenkinsci/plugins/buildmonitor/BuildMonitorView/index.jelly#L161

Current idea:
Change what the `st:bind` tag emits. Same (CSP-incompatible) behavior with no var name as with a complex/unsafe var name. That way callers with var names that are considered invalid still get their JS, just not from the new CSP-compatible path.

Reply:
https://github.com/jenkinsci/stapler/pull/385/commits/16f23e39438b731851dc446c81d32ec9c6c84589#diff-b2f036d8f4ef25d509293883bedcb5fba26848727b469d8c1d2ad4b314000203R90-R95